app/helpers/authorization.rb

```diff
@@ -0,0 +1,32 @@
+module Authorization
+
+	def authorize_or_redirect
+		redirect '/login' unless session[:user_id]
+		permissions = User.find(session[:user_id]).access_level
+		if permissions == "superadmin"
+		 	if user_id_from_params.is_a? Integer
+				redirect "/users/#{session[:user_id]}" unless user_id_from_params == session[:user_id]
+			end
+		else
+			check_member_permissions
+		end
+	end
+
+	def user_id_from_params
+		id = /users\/(\d+)\z/.match(params["splat"].join)
+		if id
+		 id[1].to_i
+		end
+	end
+
+	def check_member_permissions
```

Comment:
I'm not sure what this method does. I dont see any permissions being checked. If you need to add a layer of security over a specific route, either use a helper method at the top of each of those routes or make a before block that matched these urls.